src/python/weblyzard_api/client/ontogene/__init__.py

```diff
@@ -0,0 +1,153 @@
+    reading the data. Allows for finer handling of error codes
+    '''
+    def __init__(self, url, data):
+        self.url = url
+        #self.data = json.dumps({"hashes": data})
+        #self.headers = [{"Content-Type": "application/json"}]
+    
+    def request(self):
+        handler = urllib2.HTTPHandler()
+        opener = urllib2.build_opener(handler)
+        req = urllib2.Request(url=self.url)
+        req.add_header("Content-Type", "application/json")
+        req.get_method = lambda: "GET"
+        req.add_data(self.data)
+        try:
+            conn = opener.open(req)
+        except urllib2.HTTPError as e:
+            conn = e
+        except urllib2.URLError as e:
+            logger.error("Connection refused.. %s", e)
+            raise e
+        return conn
+
+
+class Oger(MultiRESTClient):
+    '''
+    Provides access to the OntoGene Entity Recognition.
+    
+    Currently we support the following types of requests: status, fetch and upload.
+
+    Requests to root are currently not supported since they would simply return an HTML page.
+    
+    A fetch request will retrieve an existing document from a known source:   
+        GET <BASE_URL>/fetch/<SOURCE>/<OUTPUT_FORMAT>/<DOC_ID>
+        POST <BASE_URL>/fetch/<SOURCE>/<OUTPUT_FORMAT>/<DOC_ID>
+    
+    An upload request is used to submit a text to be annotated.
+        POST <BASE_URL>/upload/<INPUT_FORMAT>/<OUTPUT_FORMAT> [/<DOC_ID>]
+    
+    Accepted values:
+        SOURCE: pubmed, pmc
+        INPUT_FORMAT: txt, bioc, pxml, nxml, pxml.gz
+        OUTPUT_FORMAT: bioc, odin, odin_custom, tsv, xml
+
+    '''
+    def __init__(self, url=OGER_API_URL, default_timeout=None):
+        '''
+        :param url: URL of the jeremia web service
+        #:param usr: optional user name
+        #:param pwd: optional password
+        '''
+        #no usr=WEBLYZARD_API_USER, pwd=WEBLYZARD_API_PASS,
+        ##user=usr, password=pwd,
+        MultiRESTClient.__init__(self, service_urls=url, default_timeout=default_timeout)
+        self.profile_cache = []
+        
+    def status(self):
+        '''
+        :returns: the status of the OGER web service.
+        '''
+        return self.request(path='status')
+
+    
+    '''
+    def fetch_document(self, docid):
+
+        fetchpath = 'fetch/pubmed/pubtator/' + str(docid)
+        print(fetchpath)
+        r = self.request(path=fetchpath)
+        
+        return r.json()
+    '''
+
+    def convert_document(self, oger_json):
+        '''
+        :returns: OGER document converted to Recognyze format.
+        '''
+        res = oger_json
+        
+        entities = {}
+        d2 = []
+        
+        #TODO: talk to OGER developers to fix this rather than have it fixed here!
+        #this is a fix for offset which always seems to be delayed with this value
+        offset = res['documents'][0]['passages'][1]['annotations'][0]['locations'][0]['offset']
```

Comment:
is this guaranteed to exist? could do with some exception handling IMO

Reply:
added error handling